### PR TITLE
[AAP-45757] - Interval Calculation Tests for RENEWAL_GUIDANCE report …

### PR DESCRIPTION
**Pull request by msmagnanijr**
[AAP-45757]

https://issues.redhat.com/browse/AAP-45757

## Description

- **What is being changed?** Added comprehensive test suite for the `get_intervals` method in `ReportRenewalGuidance` class with 6 essential tests covering core functionality, edge cases, and production patterns.

### CODE REVIEW

File: metrics_utility/test/renewal_guidance/test_get_intervals.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+        intervals_3day = basic_report_instance.get_intervals(start_date, end_date, 3)
+        intervals_7day = basic_report_instance.get_intervals(start_date, end_date, 7)
+        intervals_30day = basic_report_instance.get_intervals(start_date, end_date, 30)
+
+        assert len(intervals_3day) == 13
+        assert len(intervals_7day) == 9
+        assert len(intervals_30day) == 0
+
+        # Verify sliding window progression
```
Comment (AlexSCorey) [inline metrics_utility/test/renewal_guidance/test_get_intervals.py:145]: What is a `sliding interval progression`?

**Review by AlexSCorey — COMMENTED** [review]
I think the code here looks good.  I have confused myself on our definition of `interval` so I'd be inclined to add some definitions to our docs, if we don't have them there yet.

**Review by ShaiahWren — APPROVED** [review]
To me, this looks like a robust set of tests for the test_intervals method. And they pass :)

**Review by cshiels-ie — APPROVED** [review]

**Review by appuk — APPROVED** [review]

File: metrics_utility/test/renewal_guidance/test_get_intervals.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            'opt_ephemeral': '7 days',
+            'price_per_node': 0.1,
+            'report_period_range': '2025-01-01,2025-06-03',
+            'since_date': '2025-01-01',
+            'until_date': '2025-06-03',
+        }
+
+        return ReportRenewalGuidance(
+            dataframe=[mock_df],
+            report_period='2025-01-01,2025-06-03',
```
Comment (himdel) [inline metrics_utility/test/renewal_guidance/test_get_intervals.py:41]: Technically, `report_period` (unlike `report_period_range`) will have the value of --month, or the default for that, even when using --since, and only `report_period_range` has the right value.

But I don't think it matters here, and I'm trying to fix that anyway, https://github.com/ansible/metrics-utility/pull/136/commits/367631711c75b4fe823921f72168b9105c04a2a2, so, shouldn't be a problem :)

:+1:

**Review by himdel — APPROVED** [review]
LGTM :+1: :)

File: metrics_utility/test/renewal_guidance/test_get_intervals.py (hunk 1 above)
Comment (msmagnanijr) [reply metrics_utility/test/renewal_guidance/test_get_intervals.py:145]: In my understanding, a sliding interval progression means taking a time window of fixed duration and sliding it forward one day at a time through your data range, so you get overlapping intervals that provide continuous coverage for analysis.

File: metrics_utility/test/renewal_guidance/test_get_intervals.py (hunk 2 above)
Comment (msmagnanijr) [reply metrics_utility/test/renewal_guidance/test_get_intervals.py:41]: great :)

**Comment by msmagnanijr** [comment]
Closing in favor of https://github.com/ansible/metrics-utility/pull/138
